traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/dns/SignatureManager.java

```diff
@@ -177,29 +200,69 @@ public void run() {
 		};
 	}
 
-	private boolean hasNewKeys(final Map<String, List<DnsSecKeyPair>> keyMap, final Map<String, List<DnsSecKeyPair>> newKeyMap) {
-		for (final String key : newKeyMap.keySet()) {
-			if (!keyMap.containsKey(key)) {
-				return true;
+	private void cleanRRSIGCache(final Map<String, List<DnsSecKeyPair>> oldKeyMap, final Map<String, List<DnsSecKeyPair>> newKeyMap) {
+		synchronized (RRSIGCacheLock) {
+			if (RRSIGCache.isEmpty() || oldKeyMap == null || getKeyDifferences(oldKeyMap, newKeyMap).isEmpty()) {
+				return;
+			}
+			final int oldKeySize = RRSIGCache.size();
+			final int oldRRSIGSize = RRSIGCache.values().stream().map(Map::size).reduce(0, Integer::sum);
+			final long now = new Date().getTime();
+			final ConcurrentMap<RRSIGCacheKey, ConcurrentMap<RRsetKey, RRSIGRecord>> newRRSIGCache = new ConcurrentHashMap<>();
+			newKeyMap.forEach((name, keyPairs) -> {
+				keyPairs.forEach(keypair -> {
```

Comment:
Nit: Since there is only 1 expression within `newKeyMap.forEach()`'s lambda, this can just be

```java
			newKeyMap.forEach((name, keyPairs) -> keyPairs.forEach(keypair -> {
```

Reply:
Sure, will collapse it